Add Umbrella Network (#1698)
maticnetwork/matic-docs: docs/develop/oracles/umbrella.md

---
id: umb
title: Umbrella Network
sidebar_label: Umbrella Network
description: Umbrella Network is a decentralized oracle service that provides blockchain projects with secure, scalable, and customizable data solutions.
keywords:
  - oracle
  - umbrella
  - network
  - Layer 2
  - offchain
  - onchain
  - data
  - prices
image: https://wiki.polygon.technology/img/polygon-logo.png
---
import useBaseUrl from '@docusaurus/useBaseUrl';

:::caution Content disclaimer

Please view the third-party content disclaimer [<ins>here</ins>](https://github.com/maticnetwork/matic-docs/blob/master/CONTENT_DISCLAIMER.md).

:::

---

[<ins>Umbrella Network</ins>](https://umb.network/) is a decentralized oracle service that provides blockchain projects with secure, scalable, and customizable data solutions.

> It leverages a network of decentralized community validators that secures the reliability of the data. 

Reviewer: ```suggestion
The service leverages a network of decentralized community validators to ensure the reliability of the oracle data.
```

## Solutions   

> Umbrella Network offers 3 ways to consume data:

Reviewer: ```suggestion

Umbrella Network provides three data consumption options: on-chain data, layer 2 data, and on-demand data.
```

### On-chain data

> Read data feeds directly on-chain. Projects could define parameters such as deviation triggers and heartbeat to control the number of updates and the precission of the feeds. This is the most straightforward method to read data.

Reviewer: ```suggestion

Retrieve data feeds directly on-chain. Projects can define parameters such as deviation triggers and heartbeat to manage the frequency and precision of the updates. This is the simplest and most straightforward approach for reading data.
```

### Layer-2 data

> Read data feeds off-chain, and verify them on-chain to safely use them on your dApp. Prices are updated off-chain at a predefined frequency, and eeb3 apps fetches them as required, and submits them on-chain. This solution is more suitable if your project requires high volumes of data, where having it written on-chain is prohibitively from a cost perspective.

Reviewer: ```suggestion

Read data feeds off-chain and verify them on-chain to ensure secure utilization within your dApp. Prices are regularly updated off-chain at a predefined frequency, and web3 apps fetch this data when needed and submit it on-chain. This approach is particularly advantageous when your project necessitates handling large volumes of data, as storing it entirely on-chain would be cost-prohibitive.
```

### On-demand data

> Data is not predefined. Data will be fetch and delivered upon request of the web3 app. Planned for Q3 2023

Reviewer: ```suggestion

On-demand data fetching planned for Q3 2023.
```

> ## Technical documentation

Reviewer: ```suggestion
## Resources

### Technical documentation
```

> You can find our technical documentation [<ins>here</ins>](https://umbrella-network.readme.io/docs).

Reviewer: ```suggestion

The official technical documentation can be found [<ins>here</ins>](https://umbrella-network.readme.io/docs).
```

> ## Custom feeds

Reviewer: ```suggestion
### Custom feeds
```

> In case your project requires a custom feed, please contact us [<ins>here</ins>](https://www.umb.network/contact#form).

Reviewer: ```suggestion

In case your project requires a custom feed, please contact the Umbrella team [<ins>here</ins>](https://www.umb.network/contact#form).
```
